index.bs

```diff
@@ -1466,8 +1494,14 @@ Since the {{MLOperand/[[builder]]}} object is bound by the {{MLGraphBuilder/cons
 
 If an operation supports only a subset of {{MLOperandDataType}}s, the <dfn>allowed data types</dfn> for each of the operation's input operands, including both positional arguments and options, are given as either an explicit list of {{MLOperandDataType}}s, or a constraint that the operand's [=MLOperand/dataType=] must be the <dfn lt="same type as">same as</dfn> the [=MLOperand/dataType=] of another input operand, or <dfn lt="any data type">any</dfn> to allow any {{MLOperandDataType}}.
 
+Implementations may support additional data types for operands. This can be queried for each operation using the {{MLContext/opSupportLimits()}} method on {{MLContext}} and inspecting the {{MLDataTypeLimits/dataTypes}} value of the corresponding member for the operation.
+
+Issue: Can conforming implementations support *fewer* data types for an operation than what is listed for their [=allowed data types=]? In other words, is the list of allowed types only a recommendation?
+
 If an operation requires input operands with a particular [=MLOperand/rank=], the <dfn>allowed ranks</dfn> for each of the operation's input operands, including both positional arguments and options, are given as an explicit rank (e.g. 1), or <dfn lt="any rank">N</dfn> to allow any dimensionality, or the <dfn lt="same rank as">same as</dfn> another operand. More specific constraints are common, such as when an input operand's shape must be [=/unidirectionally broadcastable=] to or [=/bidirectionally broadcastable=] with another input operand; in these cases, the [=/allowed ranks=] are listed as a range, with specific validation given as steps in the operation.
 
+Implementations may (and usually do) impose an upper bound on the [=MLOperand/rank=] of operands. This can be queried for each operation using the {{MLContext/opSupportLimits()}} method on {{MLContext}} and inspecting the {{MLTensorLimits/rankRange}}.{{MLRankRange/max}} value of the corresponding member for the operation.
```

Comment:
And lower bound?

Reply:
A lower bound higher than what is specified as the minimum seems odd to me. Do we anticipate that happening?

Reply:
e.g. [prelu on CoreML](https://apple.github.io/coremltools/source/coremltools.converters.mil.mil.ops.defs.html#coremltools.converters.mil.mil.ops.defs.iOS15.activation.prelu) that requires minimum rank of 3? For this case I did emulate instead of exposing such rank range, but for some cases we probably prefer to not emulate but just expose such rankRange.

Reply:
I think about it more as the implementation supporting a narrower range than supported by the spec, which could include both a higher lower bound and a lower upper bound. I agree that it is more likely that an implementation is constrained on the high end. The most common example on the low end is not supporting scalars, which we can polyfill easily enough that it doesn't come up in practice.

It makes things easier for frameworks if both min and max are provided by `opSupportLimits()` even if the min value in practice is mostly defined by the spec and not overridden by the implementation.

Reply:
Done in 7328146